rust/src/transformations/make_stable_expr/expr_discrete_quantile_score/mod.rs

```diff
@@ -0,0 +1,208 @@
+
+use polars::lazy::dsl::Expr;
+use polars::prelude::DataType::*;
+
+mod plugin_dq_score;
+pub(crate) use plugin_dq_score::DQScoreArgs;
+
+/// Make a measurement that adds Laplace noise to a Polars expression.
+///
+/// # Arguments
+/// * `input_domain` - ExprDomain
+/// * `input_metric` - The metric space under which neighboring LazyFrames are compared
+/// * `expr` - The expression to which the Laplace noise will be added
+pub fn make_expr_discrete_quantile_score<MI>(
+    input_domain: ExprDomain,
+    input_metric: PartitionDistance<MI>,
+    expr: Expr,
+) -> Fallible<
+    Transformation<ExprDomain, ExprDomain, PartitionDistance<MI>, Parallel<LInfDistance<f64>>>,
+>
+where
+    MI: 'static + UnboundedMetric,
+    Expr: StableExpr<PartitionDistance<MI>, PartitionDistance<MI>>,
+    (ExprDomain, PartitionDistance<MI>): MetricSpace,
+{
+    let Some((input, args)) = match_dq_score(&expr)? else {
+        return fallible!(MakeTransformation, "Expected quantile_score function");
+    };
+
+    let DQScoreArgs {
+        alpha, candidates, ..
+    } = args;
+
+    let t_prior = input.clone().make_stable(input_domain, input_metric)?;
+    let (middle_domain, middle_metric) = t_prior.output_space();
+
+    let active_series = middle_domain.active_series()?.clone();
+    if active_series.nullable {
+        return fallible!(
+            MakeTransformation,
+            "Quantile estimation requires non-null inputs"
+        );
+    }
+
+    fn validate<T: Number + RoundCast<f64>>(candidates: &Vec<f64>) -> Fallible<()> {
+        validate_candidates(
+            &candidates
+                .iter()
+                .cloned()
+                .map(T::round_cast)
+                .collect::<Fallible<_>>()?,
+        )
+    }
+
+    match active_series.field.dtype {
+        UInt32 => validate::<u32>(&candidates),
+        UInt64 => validate::<u64>(&candidates),
+        Int8 => validate::<i8>(&candidates),
+        Int16 => validate::<i16>(&candidates),
+        Int32 => validate::<i32>(&candidates),
+        Int64 => validate::<i64>(&candidates),
+        Float32 => validate::<f32>(&candidates),
+        Float64 => validate::<f64>(&candidates),
+        UInt8 | UInt16 => {
+            return fallible!(
+                FailedFunction,
+                "u8 and u16 not supported in the OpenDP Polars plugin. Please use u32 or u64."
+            )
+        }
+        dtype => {
+            return fallible!(
+                MakeTransformation,
+                "Expected numeric data type, found {:?}",
+                dtype
+            );
+        }
+    }?;
+
+    let margin = middle_domain.active_margin()?.clone();
+
+    let mpl = margin
+        .max_partition_length
+        .ok_or_else(|| err!(MakeTransformation, "Must know max_partition_length"))?;
+
+    let constants = score_candidates_constants::<u64>(Some(mpl as u64), alpha)?;
+    let (alpha_num, alpha_den, _) = constants.clone();
```

Comment:
`den` = `density`? Might be clear to intended audience.

Reply:
denominator. Left a comment.